### PR TITLE
State of the safety critical tools

### PR DESCRIPTION
**Pull request by alexandruradovici**
This pull request adds a list of requirements and existing tools of safety critical certification. This reflects the results of the meetings of the tooling subcommittee.

@af-airbus please take a look at the DAL related entries, some of the lines in the table do not have a DAL minimum requirements
@jgust @bgenis @pellico and @jmqd please take a look at the ASIL minimum ASIL items, some of the lines of the table arte missing ASIL

### CODE REVIEW

File: subcommittee/tooling/mission-statement.md [hunk 1]
```diff
@@ -6,11 +6,29 @@
...
-| Code Metrics Generator | Cyclomatic Complexity, ... | N/A | not available |
+| Need ID | Minimum Relevant Standard Rule | Tools | Notes |
+|---------|--------------------------------|-------|-------|
+| DC Coverage | ASIL-D, DAL-B | llvm (19, rustc 1.82) | - unstable, no macros / pattern matching  |
+| MC Coverage | ASIL-C. DAL-A | llvm (19, rustc 1.82) | - AdaCore report generator plans to stabilize the feature by end of 2025 <br> - unstable, no macros / pattern matching |
+| Statement Coverage | ASIL-A, DAL-C | llvm (19, rustc 1.82), [cargo-tarpaulin](https://github.com/xd009642/tarpaulin) | - unstable, no macros / pattern matching  |
+| Function coverage | ASIL-C, DAL-C | llvm (19, rustc 1.82) - unstable, no macros / pattern matching, [cargo-tarpaulin](https://github.com/xd009642/tarpaulin) |
+| Call coverage | ASIL-C, DAL-C | llvm (19, rustc 1.82), [cargo-tarpaulin](https://github.com/xd009642/tarpaulin) | - unstable, no macros / pattern matching |
+| Branch Coverage | ASIL-B, DAL-C | llvm (19, rustc 1.82), [cargo-tarpaulin](https://github.com/xd009642/tarpaulin) | - unstable, no macros / pattern matching |
+| Qualified Compiler | ASIL-D, DAL-D  | ferroccene, Ada Core |
```
Comment (MerrimanInd) [inline subcommittee/tooling/mission-statement.md:17]: I believe the HighTec LLVM based compiler is also ASIL-D certified.

File: subcommittee/tooling/mission-statement.md [hunk 2]
```diff
@@ -6,11 +6,29 @@
...
+| Fault Injection Tests | ASIL-D, DAL-D | N/A | |
+| Test Coverage | ASIL-, DAL-D | llvm (19), cargo-test, [mantra?](https://crates.io/crates/mantra) | |
+| Underfined Behaviour Absence | ASIL-A, DAL-C | ensured by static analysis and source code confirmity |
```
Comment (MerrimanInd) [inline subcommittee/tooling/mission-statement.md:20]: I think [Miri](https://github.com/rust-lang/miri) calls itself an UB detection tool so maybe that should slot in here? Or elsewhere?

File: subcommittee/tooling/mission-statement.md [hunk 3]
```diff
@@ -6,11 +6,29 @@
...
+| Static Analysis Tools | DAL-C | | - make sure coding guidelines specify some rules, probably similar to Polyspace, <br> - an assessment about what the rust compiler covers is necessary <br> - for DO-178 such tools support the *source code conformity* objective |
+| Unambiguous Graphical Representation | ASIL-B | UML like too, but for Rust, AUTOSAR suggests - [Explanation of ARA Applications in Rust](https://www.autosar.org/fileadmin/standards/R23-11/AP/AUTOSAR_AP_EXP_ARARustApplications.pdf) |
+| Formal Verification | ASIL-C (recommended), DAL-C (recommended) | - [verifast](https://github.com/verifast/verifast) WIP <br> - [creusot](https://github.com/creusot-rs/creusot) WIP, works only for safe rust <br> - gillian-creusot WIP, not in a usable state, works for unsafe rust, [kani](https://model-checking.github.io/kani/) | - prove absence of UB <br> - prove that the code does what it should do <br> - using some spec (different language, eg SPARC or CodeProver, prove that an output is not possible) |
```
Comment (MerrimanInd) [inline subcommittee/tooling/mission-statement.md:23]: [Verus](https://github.com/verus-lang/verus) is in beta but a formal verification tool that might get thrown in here.

File: subcommittee/tooling/mission-statement.md [hunk 4]
```diff
@@ -6,11 +6,29 @@
...
+| Test runner | all | - [cargo-nextest](https://nexte.st) <br> - cargo <br> - defmt-test + probe.rs | No ISO qualified tool are available. Example of qualified test runner: [VectorCast](https://www.vector.com/us/en/products/products-a-z/software/vectorcast/?gad_source=1&gclid=EAIaIQobChMIyMKCgvn3igMVM0-RBR1s9BOkEAAYASAAEgJM2_D_BwE#c336977),  [TESSY](https://www.razorcat.com/en/product-tessy.html) <br> - tightly linked to the coverage level |
+| Automatic generation of unit/integration tests based on equivalence classes and boundary values | ASIL-B | | Example of automatic test generator based on equivalence classe and boundary values: [VectorCast](https://www.vector.com/us/en/products/products-a-z/software/vectorcast/?gad_source=1&gclid=EAIaIQobChMIyMKCgvn3igMVM0-RBR1s9BOkEAAYASAAEgJM2_D_BwE#c336977),  [TESSY](https://www.razorcat.com/en/product-tessy.html) |
```
Comment (MerrimanInd) [inline subcommittee/tooling/mission-statement.md:29]: Would property based testing like [proptest](https://github.com/proptest-rs/proptest) fit in here?

File: subcommittee/tooling/mission-statement.md (hunk 1 above)
Comment (pellico) [reply subcommittee/tooling/mission-statement.md:17]: Yes I confirm this.

Comment (pazmank) [reply subcommittee/tooling/mission-statement.md:17]: ISO 26262 certified for Tricore, released, but not certified for arm

**Comment by PLeVasseur** [comment]
Was notified of [`coq-of-rust`](https://github.com/formal-land/coq-of-rust).

I know we had the discussion last time that the Venn diagram of "safety-critical standards" and "formal methods" tend to have little to no overlap, but thought I'd mention it as it came up [here](https://www.linkedin.com/feed/update/urn:li:activity:7305239373883531265?commentUrn=urn%3Ali%3Acomment%3A%28activity%3A7305239373883531265%2C7308048996881592320%29&dashCommentUrn=urn%3Ali%3Afsd_comment%3A%287308048996881592320%2Curn%3Ali%3Aactivity%3A7305239373883531265%29).
